ci-operator/step-registry/upi/conf/vsphere/platform-external/upi-conf-vsphere-platform-external-commands.sh

```diff
@@ -0,0 +1,228 @@
+
+# install-config for `None` platform type is expected
+declare platform_none="none: {}"
+if ! grep -F "${platform_none}" "${install_config}"
+then
+        echo "install-config contains platform other than None, can not proceed"
+        exit 1
+fi
+
+#set machine cidr if proxy is enabled
+if grep 'httpProxy' "${install_config}" ; then
+  cat >> "${install_config}" << EOF
+networking:
+  machineNetwork:
+  - cidr: "192.168.${third_octet}.0/25"
+EOF
+fi
+
+echo "$(date -u --rfc-3339=seconds) - Create terraform.tfvars ..."
+cat > "${SHARED_DIR}/terraform.tfvars" <<-EOF
+machine_cidr = "192.168.${third_octet}.0/25"
+vm_template = "${vm_template}"
+vsphere_cluster = "${vsphere_cluster}"
+vsphere_datacenter = "${vsphere_datacenter}"
+vsphere_datastore = "${vsphere_datastore}"
+vsphere_server = "${vsphere_url}"
+ipam = "ipam.vmc.ci.openshift.org"
+cluster_id = "${cluster_name}"
+base_domain = "${base_domain}"
+cluster_domain = "${cluster_domain}"
+ssh_public_key_path = "${ssh_pub_key_path}"
+compute_memory = "16384"
+compute_num_cpus = "4"
+vm_network = "${LEASED_RESOURCE}"
+vm_dns_addresses = ["${dns_server}"]
+bootstrap_ip_address = "192.168.${third_octet}.3"
+lb_ip_address = "192.168.${third_octet}.2"
+compute_ip_addresses = ["192.168.${third_octet}.7","192.168.${third_octet}.8","192.168.${third_octet}.9"]
+control_plane_ip_addresses = ["192.168.${third_octet}.4","192.168.${third_octet}.5","192.168.${third_octet}.6"]
+EOF
+
+echo "$(date -u --rfc-3339=seconds) - Create secrets.auto.tfvars..."
+cat > "${SHARED_DIR}/secrets.auto.tfvars" <<-EOF
+vsphere_password="${GOVC_PASSWORD}"
+vsphere_user="${GOVC_USERNAME}"
+ipam_token=""
+EOF
+
+dir=/tmp/installer
+mkdir "${dir}/"
+pushd ${dir}
+cp -t "${dir}" \
+    "${SHARED_DIR}/install-config.yaml"
+
+echo "$(date +%s)" > "${SHARED_DIR}/TEST_TIME_INSTALL_START"
+
+### Create manifests
+echo "Creating manifests..."
+openshift-install --dir="${dir}" create manifests &
+
+set +e
+wait "$!"
+ret="$?"
+set -e
+
+if [ $ret -ne 0 ]; then
+  cp "${dir}/.openshift_install.log" "${ARTIFACT_DIR}/.openshift_install.log"
+  exit "$ret"
+fi
+
+# remove channel from CVO
+sed -i '/^  channel:/d' "manifests/cvo-overrides.yaml"
+
+### Remove control plane machines
+echo "Removing control plane machines..."
+rm -f openshift/99_openshift-cluster-api_master-machines-*.yaml
+
+### Remove compute machinesets (optional)
+echo "Removing compute machinesets..."
+rm -f openshift/99_openshift-cluster-api_worker-machineset-*.yaml
+
+### Update platform spec in the infra resource
+echo "Patching infrastructure resource manifest with the 'External' platform type..."
+cat <<EOF > /tmp/infraPlatformSpecPatch.yml
+spec:
+  platformSpec:
+    type: External
+    external:
+      platformName: vSphere
+status:
+  platform: External
+  platformStatus:
+      type: External
+      external: {}
+EOF
+yq-go m -x -i "manifests/cluster-infrastructure-02-config.yml" "/tmp/infraPlatformSpecPatch.yml"
```

Comment:
could just the manifests be modified in a step following upi-conf-vsphere instead of replicating the bulk of the step?  i'm not necessarily opposed to replicating if we need to.

Reply:
This will require splitting `upi-conf-vsphere` step and passing all manifests between steps.
Now, `upi-conf-vsphere` generates manifests and uses them instantly for generating ignition files, manifests get deleted then. 

I will try to do so, but not sure how many jobs it might affect.

Reply:
no worries @lobziik , i was just curious